test/unit/transpiler/passes/scheduling/test_dynamical_decoupling.py

```diff
@@ -693,6 +683,51 @@ def test_reproducible(self):
 
         self.assertEqual(qc_dd0, qc_dd1)
 
+    def test_nested_block_dd(self):
+        """Test DD applied within a block."""
+
+        dd_sequence = [XGate(), XGate()]
+        pm = PassManager(
+            [
+                ASAPScheduleAnalysis(self.durations),
+                PadDynamicalDecoupling(
+                    self.durations,
+                    dd_sequence,
+                    pulse_alignment=1,
+                    sequence_min_length_ratios=[0.0],
+                ),
+            ]
+        )
+
+        qc = QuantumCircuit(3, 1)
+        qc.x(1)
+        qc.x(2)
+        qc.barrier()
+        with qc.if_test((0, True)):
+            qc.delay(1000, 1)
+
+        qc_dd = pm.run(qc)
+
+        expected = QuantumCircuit(3, 1)
+        expected.delay(50, 0)
+        expected.x(1)
+        expected.x(2)
+        expected.barrier()
+        with expected.if_test((0, True)):
+            expected.delay(225, 1)
+            expected.x(1)
+            expected.delay(450, 1)
+            expected.x(1)
+            expected.delay(225, 1)
+            expected.delay(225, 2)
+            expected.x(2)
+            expected.delay(450, 2)
+            expected.x(2)
+            expected.delay(225, 2)
+            expected.delay(1000, 0)
```

Comment:
This is because the qubit has not yet been touched., ie., it is assumed to be in the `|0>` state. This may be partially controlled with `skip_reset_qubits` and is behaviour inherited from the standard Qiskit implementation.